### ISSUE TITLE
tahoe: Consider use of subdomain

### ISSUE THREAD

**Issue by jvalleroy** [freedombox/Plinth#874]
Current the Tahoe-LAFS web interface is provided on port 5678. Consider using a subdomain instead.

**Comment by mray** [comment]
Are there any best practices to follow generally? If it makes sense to generally recommend one way over the other FreedomBox should encourage that among all other modules.

**Comment by JosephKiranBabu** [comment]
@jvalleroy How about this? Every time a domain is selected for an application, like for Matrix or Diaspora, we can check whether <subdomain>.domain.tld has a Lets Encrypt certificate or obtain one automatically if it doesn't.
@mray If we have the feature of automatically obtaining Lets Encrypt certificates, we can recommend getting subdomains over other methods like setting ports. Maybe we should do that. 
Sunil had some more ideas on subdomains.

**Comment by JoKeyser** [comment]
What I don't like about using subdomains is that it generally involves setting up another subdomain in your domain provider, or am I wrong? For the domains I own, I think I would have to set up another DNS entry - and it takes a while (up to 1 day) to trickle down until it becomes usable...

**Comment by JosephKiranBabu** [comment]
Once Let's Encrypt has wildcard certificate support (maybe in 2018), we wouldn't have to worry about certificates for subdomains anymore.
http://www.zdnet.com/article/lets-encrypt-brings-wildcard-certificates-to-the-web/

Till then, we'll have to keep obtaining certificates for every new domain added.

**Comment by jvalleroy** [comment]
I think @JoKeyser's question was about getting the subdomain pointed to the right place in DNS, not about the certificate.
- Does this always need to be done through the domain name provider?
- Is it typically available for free domain names?
- Does running an authoritative DNS server help here...I guess only if you are directing your DNS queries to the right place?
- Can it work with pagekite or onion address?

**Comment by JoKeyser** [comment]
@joelvalleroy , yep, thanks, that's what I meant, and @JosephKiranBabu thanks for the reminder of wildcard certs.

For simplicity (of implementation *and* end-user understanding), I'm currently very much in favor of offering *all* apps at a subfolder of the apps name.
Asking provocatively: Is there really any benefit of using subdomains (in our use-case), that makes the added complexity worthwhile?
